components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/impl/APIUsageStatisticsRestClientImpl.java

```diff
@@ -1408,11 +1392,19 @@ public static OMElement buildOMElement(InputStream inputStream) throws Exception
         //extending lucene query with time ranges
         try {
             query = APIUsageStatisticsClientConstants.REQUEST_TIME + ": [" + RestClientUtil.getFloorDateAsLong(fromDate)
-                    + " TO " + RestClientUtil.getCeilingDateAsLong(toDate) + "]";
+                    + " TO " + RestClientUtil.getCeilingDateAsLong(toDate) + ']';
         } catch (ParseException e) {
             handleException("Error occurred while Error parsing date", e);
         }
 
+        //if My APIs stat, add constraint with API publisher
+        if (!providerName.equals(APIUsageStatisticsClientConstants.ALL_PROVIDERS)) {
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/9619da2271b49f37ef93456b6f8a8ac235f685ea